wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift

```diff
@@ -384,6 +396,27 @@ public final class ZMUserSession: NSObject {
         self.contextStorage = contextStorage
         self.recurringActionService = recurringActionService
         self.dependencies = dependencies
+
+        super.init()
+
+        setupAnalyticsSession()
```

Comment:
An initializer should only initialize and not trigger side effects such as starting an analytics session. I suggest to create the analytics session, and only start it when `setup` is called.

Reply:
Done: https://github.com/wireapp/wire-ios/commit/7caa26a9f21c0b50fca5b69418abbd477f28e8ac

Reply:
Maybe you could move also `setupAnalyticsSession` inside the `setUp` method.